label_studio/blueprint.py

```diff
@@ -0,0 +1,1085 @@
+
+
+@blueprint.route('/data/<path:filename>')
+@requires_auth
+@exception_treatment
+def get_data_file(filename):
+    """ External resource serving
+    """
+    project = project_get_or_create()
+
+    # support for upload via GUI
+    if filename.startswith('upload/'):
+        path = os.path.join(project.name, filename)
+        directory = os.path.abspath(os.path.dirname(path))
+        filename = os.path.basename(path)
+        return flask.send_from_directory(directory, filename, as_attachment=True)
+
+    # serving files from local storage
+    if not project.config.get('allow_serving_local_files'):
+        raise FileNotFoundError('Serving local files is not allowed. '
+                                'Use "allow_serving_local_files": true config option to enable local serving')
+    directory = request.args.get('d')
+    return flask.send_from_directory(directory, filename, as_attachment=True)
+
+
+def str2datetime(timestamp_str):
+    try:
+        ts = int(timestamp_str)
+    except:
+        return timestamp_str
+    # return datetime.utcfromtimestamp(ts).strftime('%Y%m%d.%H%M%S')
+    return datetime.utcfromtimestamp(ts).strftime('%c')
+
+
+def start_browser(ls_url, no_browser):
+    import threading
+    import webbrowser
+    if no_browser:
+        return
+
+    browser_url = ls_url + '/welcome'
+    threading.Timer(2.5, lambda: webbrowser.open(browser_url)).start()
+    print('Start browser at URL: ' + browser_url)
+
+
+def main():
+    input_args = parse_input_args()
+    app = create_app(LabelStudioConfig(input_args=input_args), set_str2datetime=True)
+
+    # setup logging level
+    if input_args.log_level:
+        logging.root.setLevel(input_args.log_level)
+
+    # On `init` command, create directory args.project_name with initial project state and exit
+    if input_args.command == 'init':
+        Project.create_project_dir(input_args.project_name, input_args)
+        return
+
+    elif input_args.command == 'start':
+
+        # If `start --init` option is specified, do the same as with `init` command, but continue to run app
+        if input_args.init:
+            Project.create_project_dir(input_args.project_name, input_args)
+
+        if not os.path.exists(Project.get_project_dir(input_args.project_name, input_args)):
+            raise FileNotFoundError(
+                'Project directory "{pdir}" not found. '
+                'Did you miss create it first with `label-studio init {pdir}` ?'.format(
+                    pdir=Project.get_project_dir(input_args.project_name, input_args)))
+
+    # On `start` command, launch browser if --no-browser is not specified and start label studio server
+    if input_args.command == 'start':
+        import label_studio.utils.functions
+        import label_studio.utils.auth
+        config = Project.get_config(input_args.project_name, input_args)
+
+        # set username and password
+        label_studio.utils.auth.USERNAME = input_args.username or \
```

Comment:
I'd like to change this to get the username and password from  `current_app.label_studio.username` and `current_app.label_studio.password`